File: Ejercicio 7/figuras.py
from abc import ABC,abstractmethod
import math

class Calculable(ABC):
    @abstractmethod
    def area(self): pass

class Figura(ABC):
    @abstractmethod
    def nombre(self): pass

class Circulo(Figura,Calculable):
    def __init__(self,r): self.r=r
    def nombre(self): print("Circulo")
    def area(self): return math.pi*self.r*self.r

class Rectangulo(Figura,Calculable):
    def __init__(self,a,b):
        self.a=a
        self.b=b
    def nombre(self): print("Rectangulo")
    def area(self): return self.a*self.b

class Triangulo(Figura,Calculable):
    def __init__(self,b,h):
        self.b=b
        self.h=h
    def nombre(self): print("Triangulo")
    def area(self): return (self.b*self.h)/2

figuras=[Circulo(3),Rectangulo(4,5),Triangulo(6,2)]

for f in figuras:
    print("Area:",f.area())

File: Ejercicio 7/figuras.java
interface Calculable{
    double area();
}

abstract class Figura{
    abstract void nombre();
}

class Circulo extends Figura implements Calculable{
    double r;

    Circulo(double r){ this.r=r; }

    void nombre(){ System.out.println("Circulo"); }

    public double area(){
        return Math.PI*r*r;
    }
}

class Rectangulo extends Figura implements Calculable{
    double a,b;

    Rectangulo(double a,double b){
        this.a=a;
        this.b=b;
    }

    void nombre(){ System.out.println("Rectangulo"); }

    public double area(){
        return a*b;
    }
}

class Triangulo extends Figura implements Calculable{
    double b,h;

    Triangulo(double b,double h){
        this.b=b;
        this.h=h;
    }

    void nombre(){ System.out.println("Triangulo"); }

    public double area(){
        return (b*h)/2;
    }
}

public class Main{
    public static void main(String[] args){

        Calculable[] figuras={
            new Circulo(3),
            new Rectangulo(4,5),
            new Triangulo(6,2)
        };

        for(Calculable f:figuras)
            System.out.println("Area: "+f.area());
    }
}
